File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/1895-minimum-number-of-operations-to-move-all-balls-to-each-box.js
/**
 * @param {string} boxes
 * @return {number[]}
 */
var minOperations = function (boxes) {
    let n = boxes.length;
    let b = 0, m = 0;
    const res = Array(n).fill(0);

    for (let i = 0; i < n; i++) {
        res[i] = (b + m);
        m += b;
        b += (boxes.charCodeAt(i) - 48);
    }

    b = m = 0;
    for (let i = n - 1; i >= 0; i--) {
        res[i] += (b + m);
        m += b;
        b += (boxes.charCodeAt(i) - 48);
    }
    return res;
};

File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/1895-minimum-number-of-operations-to-move-all-balls-to-each-box.go
func minOperations(boxes string) []int {
    n := len(boxes)

    res := make([]int, n)
    b, m := 0, 0
    for i := 0; i < n; i++ {
        res[i] = (b + m)
        m += b
        b += int(boxes[i] - '0')
    }
    b = 0
    m = 0
    for i := n - 1; i >= 0; i-- {
        res[i] += (b + m)
        m += b
        b += int(boxes[i] - '0')
    }
    return res
}